### ISSUE TITLE
bug: Skyra crashes when attempting to delete messages that can not be deleted.

### ISSUE THREAD

**Issue by GlebTiK** [skyra-project/skyra#2697]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

Whenever a user attempts to purge a single message that is undeletable (in this case, the message is "\<User\> changed the channel name: \<new thread name\>" as seen on the screenshot) it will cause the bot to crash fully and will require (at least I believe so) a manual restart.
If there are any messages before or after the "\<User\> changed the channel name: \<new thread name\>" Skyra will not crash and will work as intended: it will clear all messages that are deletable.
The "code" of the error: 0f80f4f3140c41dca2dad42616a6e4cd

### Steps To Reproduce

1. Create a thread in any channel Skyra has access to (Skyra must have access to the thread too).
2. Ping Skyra in the thread so that Skyra starts reading all the messages that are in the thread.
3. Remove all the messages from everyone in the thread (including from the user who created the thread and Skyra), making it empty.
4. Rename the thread to something that does not match the previous name.
5. Use the purge command right after renaming the thread.

### Expected behavior

Skyra responding after the commands and not shutting down, or an error message saying there is nothing to delete.

### Screenshots

![Screenshot of Skyra not responding](https://github.com/user-attachments/assets/452642ec-2060-42ed-916e-9bea3b032504)

**Comment by GlebTiK** [comment]
I believe that the code near the following line needs to be edited in order for this to be fixed: https://github.com/skyra-project/skyra/blob/9e63ba50870d8917c06d0905a8f577228c7ffab8/src/commands/Moderation/prune.ts#L85

**Comment by kyranet** [comment]
There isn't actually a crash, more like an infinite loop somehow that I have been trying to find for around a month 😅

I recently made #2690 to address this issue, but hasn't been deployed yet, I just deployed the latest version of main (9e63ba50870d8917c06d0905a8f577228c7ffab8) to Skyra Beta now, can you check if the issue still happens? https://discord.com/oauth2/authorize?client_id=1274279625005993994

This is with Skyra Beta just now:
![image](https://github.com/user-attachments/assets/4cf62679-5edb-42e1-8f29-8562e57a1d2c)

**Comment by GlebTiK** [comment]
Yup, can confirm that the issue is fixed on Beta.
I will be closing this issue now, as it's now a matter of time until this gets added to stable version of Skyra.
![Skyra responding and functioning as intended](https://github.com/user-attachments/assets/52113dab-f12c-48e8-a91e-8ce3d5fc620f)
By the way, thanks for creating such a cool bot! I love open source bots that are simple like these :)

**Comment by kyranet** [comment]
I can't thank you enough for giving me the ultimate proof I needed to find the issue... a DoS bug with an integration in our reporter tool, external to us.

I made a workaround in the meantime while they fix that, and I'll release that.

Usually we wouldn't do releases this late in the night, specially when the next day is a working day, but since it's related to a DoS... and the changes between latest production versions are small, I think it's safe. I'll still keep a watch on it while I try to do the things I need to do before sleeping and waking up in less than 8 hours 😅
